src/py21cmfast/_utils.py

```diff
@@ -939,7 +939,13 @@ def exists(self, direc=None):
         """
         return self.find_existing(direc) is not None
 
-    def write(self, direc=None, fname=None, write_inputs=True, mode="w"):
+    def write(
+        self,
+        direc=None,
+        fname=str | Path | None | h5py.File | h5py.Group,
```

Comment:
This is problematic for python < 3.10 and some tests seem to be failing because of it.
It should be something like: `fname : Union[str, Path, None, h5py.File, h5py.Group] = None`